Description: Add aspell support based on code from old aspell-de-alt package.
Author: Roland Rosenfeld <roland@debian.org>
Forwarded: no upstream available

--- a/Makefile
+++ b/Makefile
@@ -5,6 +5,19 @@ WORTE =	worte.txt verben.txt adjektive.t
 	geographie.txt vornamen.txt abkuerz.txt imperat.txt latein.txt \
 	informatik.txt infoabk.txt
 
+ASPELL = /usr/bin/aspell --local-data-dir=. -l de-1901
+ISPELL = /usr/bin/ispell
+PREZIP = /usr/bin/prezip
+DESTDIR =
+
+dictdir = /usr/lib/aspell
+datadir = /usr/lib/aspell
+
+cwl_files = de-common-1901.cwl de_DE-only-1901.cwl de_CH-only-1901.cwl
+data_files = de-1901.dat de-1901_phonet.dat de-1901_affix.dat
+multi_files = de_DE-1901.multi de_CH-1901.multi de-1901.alias \
+        deutsch-alt.alias german-old.alias
+
 deutsch.hash: all.words deutsch.aff
 	buildhash all.words deutsch.aff $@
  
@@ -19,13 +32,51 @@ myspelldic: deutsch.hash
 	rm myspell.dic.tmp
 
 isowordlist: deutsch.hash
-	ispell -e -d ./deutsch <all.words |tr " " "\n" | ./dic2iso | sort -u >isowordlist
+	ispell -e -d ./deutsch <all.words |tr " " "\n" | sed -f dic2iso \
+		| sort -u > isowordlist
 
 clean:
 	rm -f all.words all.words.cnt all.words.stat deutsch.hash hk*deutsch.tar.gz *.tmp myspell.dic isowordlist all.stat
+	rm -f szlig noszlig *.wl *.cwl de-1901_affix.dat
 
 sort:
 	for i in *.txt;do sort -u -o $$i $$i;done
 
 dist:	clean sort
 	tar cvf - * | gzip -9 > hk2-deutsch.tar.gz
+
+#
+# aspell stuff:
+#
+aspell: $(cwl_files) de-1901_affix.dat
+
+de-1901_affix.dat: myspell.aff
+	cp $< $@
+
+szlig: isowordlist
+	grep -a ß $< > $@
+
+noszlig: isowordlist
+	grep -av ß $< > $@
+
+de-common-1901.wl: noszlig de-1901_phonet.dat de-1901_affix.dat
+	$(ASPELL) munch-list < $< | sort -u > $@
+
+de_DE-only-1901.wl: szlig de-1901_phonet.dat de-1901_affix.dat
+	$(ASPELL) munch-list < $< | sort -u > $@
+
+de_CH-only-1901.wl: szlig de-1901_phonet.dat de-1901_affix.dat
+	sed 's/ß/ss/g' < $< | $(ASPELL) munch-list | sort -u > $@
+
+%.cwl: %.wl
+	$(PREZIP) < $< > $@
+
+aspell-install: aspell
+	mkdir -p ${DESTDIR}${dictdir}/
+	cp ${multi_files} ${DESTDIR}${dictdir}/
+	cd ${DESTDIR}${dictdir}/ && chmod 644 ${multi_files}
+	mkdir -p ${DESTDIR}${datadir}/
+	cp ${data_files} ${DESTDIR}${datadir}/
+	cd ${DESTDIR}${datadir}/ && chmod 644 ${data_files}
+
+.PHONY: aspell clean aspell-install
--- /dev/null
+++ b/deutsch-alt.alias
@@ -0,0 +1 @@
+add de_DE-1901.multi
--- /dev/null
+++ b/german-old.alias
@@ -0,0 +1 @@
+add de_DE-1901.multi
--- /dev/null
+++ b/de-1901.dat
@@ -0,0 +1,4 @@
+name       de-1901
+charset    iso-8859-1
+soundslike de-1901
+affix      de-1901
--- /dev/null
+++ b/de-1901_phonet.dat
@@ -0,0 +1,497 @@
+#   German phonetic transformation rules for use with Aspell
+#   Copyright (C) 2000 Björn Jacke
+#
+#   This library is free software; you can redistribute it and/or
+#   modify it under the terms of the GNU Lesser General Public
+#   License version 2.1 as published by the Free Software Foundation;
+#
+#   This library is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#   Lesser General Public License for more details.
+#
+#   You should have received a copy of the GNU Lesser General Public
+#   License along with this library; if not, write to the Free Software
+#   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+#   Björn Jacke may be reached by email at bjoern.jacke@gmx.de
+#
+#   Changelog:
+#
+#   2000-01-05  Björn Jacke <bjoern.jacke@gmx.de>
+#               Initial Release
+#   2000-01-07  Kevin Atkinson <kevinatk@home.com>
+#               Converted from header to data file.
+
+version         0.20
+followup        1
+collapse_result 1
+
+ÄER-                 E
+ÄU<                  EU
+Ä<                   E
+É                    E
+ÖER-                 Ö
+Ö                    Ö
+ÜBER^^               IPA
+ÜER-                 I
+Ü                    I
+ß                    Z
+ABELLE$              APL
+ABELL$               APL
+ABIENNE$             APIN
+ACEY$                AZI
+AEU<                 EU
+AE2                  E
+AGNI-^               AKN
+AGNIE-               ANI
+AGN(AEOU)-$          ANI
+AIA2                 AIA
+AIE$                 E
+AILL(EOU)-           ALI
+AINE$                EN
+AIRE$                ER
+AIR-                 E
+AISE$                EZ
+AISSANCE$            EZANZ
+AISSE$               EZ
+AIX$                 EX
+AJ(AÄEIOÖUÜ)--       A
+AKTIE                AXIE
+ALO(IY)^             ALUI
+AMATEU(RS)-          ANATÖ
+ANIELLE$             ANIL
+ANTI^^               ANTI
+ANVER^^              ANFA
+ATIA$                ATIA
+ATIA(NS)--           ATI
+ATI(AÄOÖUÜ)-         AZI
+AUAU--               _
+AUER<                AUA
+AUF^^                AUF
+AULT$                U
+AUSSE$               UZ
+AUS(ST)-^            AUZ
+AUS^^                AUZ
+AUTO^^               AUTU
+AUX(IY)-             AUX
+AUX                  U
+AU                   AU
+AVIER$               AFIE
+AYER--<              EI
+AY(AÄEIOÖUÜ)--       A
+A(IJY)<              EI
+A                    A
+BEA(BCMNRU)-^        PEA
+BEAT(AEIMORU)-^      PEAT
+BEIGE^$              PEZ
+BE(LMNRST)-^         PE
+BETTE$               PET
+BIC$                 PIZ
+BOWL(EI)-            PUL
+BP(AÄEIOÖRUÜY)-      P
+BUDGET7              PIKE
+BUFFET7              PIFE
+BYLLE$               PILE
+BYLL$                PIL
+BYTE<                PEIT
+B                    P
+CÄ-                  Z
+CÜ$                  ZI
+CACH(EI)-^           KEZ
+CAE--                Z
+CA(IY)$              ZEI
+CCH                  Z
+CCE-                 X
+CE(EIJUY)--          Z
+CENT<                ZENT
+CERST(EI)----^       KE
+CER$                 ZA
+CE3                  ZE
+CHAO(ST)-            KAU
+CHAMPIO-^            ZENPI
+CHAR(AI)-^           KAR
+CHAU(CDFSVWXZ)-      ZU
+CHE(CF)-             ZE
+CHEM-^               KE
+CHEQUE<              ZEK
+CHI(CFGPVW)-         ZI
+CH(AEUY)-<^          Z
+CHK-                 _
+CH(LOR)-<^           K
+CHST-                X
+CH(SßXZ)3            X
+CH                   K
+CIER$                ZIE
+CYB-^                ZEI
+CY9^                 ZI
+C(IJY)-3             Z
+CKST                 XT
+CK(SßXZ)3            X
+C(CK)-               _
+CLAUDET---           KLU
+CLAUDINE^$           KLUTIN
+COLE$                KUL
+COUCH                KAUZ
+CQUES$               K
+CQUE                 K
+CREAT-^              KREA
+CST                  XT
+CS<^                 Z
+C(SßX)               X
+CT(SßXZ)             X
+CZ<                  Z
+C<                   K
+D'H^                 T
+D'S3$                Z
+DAVO(NR)-^$          TAFU
+DD(SZ)--<            _
+DEPOT7               TEPU
+DESIGN               TIZEIN
+DE(LMNRST)-3^        TE
+DETTE$               TET
+DIC$                 TIZ
+DJ(AEIOU)-^          I
+DS(CH)--<            T
+DST                  ZT
+DT-                  _
+DUIS-^               TI
+DURCH^^              TURK
+DZS(CH)--            T
+D(SßZ)               Z
+D                    T
+EAULT$               U
+EAUX$                U
+EAU                  U
+EAV                  IF
+EA(AÄEIOÖÜY)-3       EA
+EA3$                 EA
+EA3                  I
+EBEN^^               EPN
+EE9                  E
+EIEI--               _
+EIH--                E
+EILLE$               EI
+EI                   EI
+EJ$                  EI
+EL-^                 E
+EL(DKL)--1           E
+EL(MNT)--1$          E
+ELYNE$               ELINE
+ELYN$                ELIN
+EL(AÄEIOÖUÜY)-1      EL
+EL-1                 L
+EM-^                 E
+EM(DFKMPQT)--1       E
+EM(AÄEIOÖUÜY)--1     E
+EM-1                 N
+EN-^                 E
+EN(CDGKQT)--1        E
+ENZ(AEIOUY)--1       EN
+EN(AÄEINOÖUÜY)-1     EN
+EN-<1                N
+ERH(AÄEIOÖUÜ)-^      ER
+ER-^                 E
+ER(AÄEIOÖUÜY)-1      A
+ER1$                 A
+ER<1                 A
+ETI(AÄOÖÜU)-         EZI
+EUEU--               _
+EUILLE$              Ö
+EUR$                 ÖR
+EUX                  Ö
+EUYS$                EUZ
+EU                   EU
+EYER<                EIA
+EY<                  EI
+E                    E
+FANS--^$             FE
+FAN-^$               FE
+FAULT-               FUL
+FEE(DL)-             FI
+FEHLER               FELA
+FE(LMNRST)-3^        FE
+FOND7                FUN
+FRAIN$               FRA
+FRISEU(RS)-          FRIZÖ  # x
+F                    F
+G'S$                 X
+GAGS^$               KEX
+GAG^$                KEK
+GD                   KT
+GEGEN^^              KEKN
+GE(LMNRST)-3^        KE
+GETTE$               KET
+G(CK)-               _
+GG-                  _
+GI(AO)-^             I
+GION$                KIUN
+GIUS-^               IU
+GMBH^$               GMPH
+GNAC$                NIAK
+GNON$                NIUN
+GN$                  N
+GONCAL-^             KUNZA
+GS(CH)--             K
+GST                  XT
+G(SßXZ)              X
+GUCK-                KU
+GUI-^                K
+G                    K
+HEAD-                E
+HE(LMNRST)-3^        E
+HE(LMN)-1            E
+HEUR1$               ÖR
+H^                   _
+IEC$                 IZ
+IEI-3                _
+IELL3                IEL
+IENNE$               IN
+IERRE$               IER
+IETTE$               IT
+IEU                  IÖ
+IE<4                 I
+IGHT3$               EIT
+IGNI(EO)-            INI
+IGN(AEOU)-$          INI
+IJ(AOU)-             I
+IJ$                  I
+IJ<                  EI
+IKOLE$               IKUL
+ILLAN(STZ)--         ILIA
+ILLAR(DT)--          ILIA
+INVER-               INFE
+ITI(AÄOÖUÜ)-         IZI
+IVIER$               IFIE
+I                    I
+JAVIE---<^           ZA
+JEAN^$               IA
+JEAN-^               IA
+JER-^                IE
+JE(LMNST)-           IE
+JOR(GK)^$            IÖRK
+J                    I
+KC(ÄEIJ)-            X
+KE(LMNRST)-3^        KE
+KH<^                 K
+KIC$                 KIZ
+KLE(LMNRST)-3^       KLE
+KOTELE-^             KUTL
+KREAT-^              KREA
+KST                  XT
+K(SßXZ)              X
+KTI(AIOU)-3          XI
+KT(SßXZ)             X
+K                    K
+LARVE-               LARF
+LEAND-^              LEAN
+LEL-                 LE
+LE(MNRST)-3^         LE
+LETTE$               LET
+LFGNAG-              LFKAN
+LIC$                 LIZ
+LIVE^$               LEIF
+LUI(GS)--            LU
+L                    L
+MASSEU(RS)-          NAZÖ
+MAURICE              NURIZ
+MBH^$                MPH
+MB(SßZ)-             N
+MC9^                 NK
+MEMOIR-^             NENUA
+ME(LMNRST)-3^        NE
+MIGUEL               NIKL
+MIKE^$               NEIK
+MN                   N
+MPJUTE-              NPUT
+MP(SßZ)-             N
+MP(BDJLMNPQRTVW)-    NP
+M                    N
+NACH^^               NAK
+NADINE               NATIN
+NAIV--               NA
+NAISE$               NEZE
+NCOISE$              ZUA
+NCOIS$               ZUA
+NEBEN^^              NEPN
+NE(LMNRST)-3^        NE
+NEN-3                NE
+NETTE$               NET
+NG(BDFJLMNPQRTVW)-   NK
+NICHTS^^             NIX
+NICHT^^              NIKT
+NINE$                NIN
+NON^^                NUN
+NOT^^                NUT
+NTI(AIOU)-3          NZI
+NTIEL--3             NZI
+NYLON                NEILUN
+ND(SßZ)$             NZ
+NT(SßZ)$             NZ
+ND'S$                NZ
+NT'S$                NZ
+NSTS$                NZ
+N                    N
+OBER^^               UPA
+OE2                  Ö
+OGNIE-               UNI
+OGN(AEOU)-$          UNI
+OIE$                 Ö
+OIR$                 UAR
+OIX                  UA
+OI<3                 EU
+OJ(AÄEIOÖUÜ)--       U
+OKAY^$               UKE
+OLYN$                ULIN
+OTI(AÄOÖUÜ)-         UZI
+OUI^                 FI
+OUILLE$              ULIE
+OU(DT)-^             AU
+OUSE$                AUZ
+OUT-                 AU
+OU                   U
+OWS$                 UZ
+OY(AÄEIOÖUÜ)--       U
+O(JY)<               EU
+O                    U
+PATIEN--^            PAZI
+PENSIO-^             PANZI
+PE(LMNRST)-3^        PE
+PFER-^               FE
+P(FH)<               F
+POLY^^               PULI
+PORTRAIT7            PURTRE
+PP(FH)--<            P
+PP-                  _
+PRIX^$               PRI
+P(SßZ)^              Z
+PTI(AÄOÖUÜ)-3        PZI
+PIC^$                PIK
+P                    P
+QUE(LMNRST)-3        KFE
+QUE$                 K
+QUI(NS)$             KI
+QU                   KF
+Q<                   K
+RCH                  RK
+RECHERCH^            REZAZ
+RER$                 RA
+RE(MNR)-4            RE
+RETTE$               RET
+RH<^                 R
+RJA(MN)--            RI
+RTI(AÄOÖUÜ)-3        RZI
+RY(KN)-$             RI
+R                    R
+SAFE^$               ZEIF
+SAUCE-^              ZUZ
+SCHSCH---7           _
+SCHTSCH              Z
+SC(HZ)<              Z
+SC                   ZK
+SELBSTST--7^^        ZELP
+SELBST7^^            ZELPZT
+SERVICE7^            ZÖRFIZ
+SE(LMNRST)-3^        ZE
+SETTE$               ZET
+SHP-^                Z
+SHST                 ZT
+SHTSH                Z
+SHT                  Z
+SH3                  Z
+SIEGLI-^             ZIKL
+SIGLI-^              ZIKL
+SIGHT                ZEIT
+SIGN                 ZEIN
+SKI(NPZ)-            ZKI
+SKI<^                ZI
+SOUND-               ZAUN
+STAATS^^             ZTAZ
+STADT^^              ZTAT
+START^^              ZTART
+STAURANT7            ZTURAN
+STEAK-               ZTE
+STRAF^^              ZTRAF
+ST'S$                Z
+STST--               _
+STS(ACEHIOUÄÜÖ)--    ZT
+ST(SZ)               Z
+STYN(AE)-$           ZTIN
+ST                   ZT
+SZE(NPT)-^           ZE
+SZI(ELN)-^           ZI
+SZCZ<                Z
+SZT<                 ZT
+SZ<3                 Z
+S                    Z
+T'S3$                Z
+TCH                  Z
+TEAT-^               TEA
+TE(LMNRST)-3^        TE
+TH<                  T
+TIC$                 TIZ
+TOAS-^               TU
+TOILET-              TULE
+TOIN-                TUA
+TRAINI-              TREN
+TSCH                 Z
+TSH                  Z
+TST                  ZT
+T(Sß)                Z
+TT(SZ)--<            _
+TT9                  T
+TZ-                  _
+T                    T
+UEBER^^              IPA
+UE2                  I
+UIE$                 I
+UM^^                 UN
+UNTERE--             UNTE
+UNTER^^              UNTA
+UNVER^^              UNFA
+UN^^                 UN
+UTI(AÄOÖUÜ)-         UZI
+U                    U
+VACL-^               FAZ
+VAC$                 FAZ
+VEDD-^               FE
+VEREIN               FAEIN
+VERSEN^              FAZN
+VER^^                FA
+VER                  FA
+VET(HT)-^            FET
+VETTE$               FET
+VIC$                 FIZ
+VIEL                 FIL
+VIEW                 FIU
+VOR^^                FUR
+VY9^                 FI
+V<                   F
+WE(LMNRST)-3^        FE
+WIC$                 FIZ
+WIEDER^^             FITA
+WY9^                 FI
+W                    F
+XE(LMNRST)-3^        XE
+X<^                  Z
+X(CSZ)               X
+XTS(CH)--            XT
+XT(SZ)               Z
+X                    X
+YE(LMNRST)-3^        IE
+YE-3                 I
+YOR(GK)^$            IÖRK
+Y(AOU)-<7            I
+YVES^$               IF
+YVONNE^$             IFUN
+Y                    I
+ZC(AOU)-             ZK
+ZE(LMNRST)-3^        ZE
+ZH<                  Z
+ZS(CHT)--            _
+ZS                   Z
+ZUERST               ZUERZT
+ZURÜCK^^             ZURIK
+ZUVER^^              ZUFA    # x
+Z                    Z
+
--- /dev/null
+++ b/de_CH-1901.multi
@@ -0,0 +1,2 @@
+add de-common-1901.rws
+add de_CH-only-1901.rws
--- /dev/null
+++ b/de_DE-1901.multi
@@ -0,0 +1,2 @@
+add de-common-1901.rws
+add de_DE-only-1901.rws
--- /dev/null
+++ b/de-1901.alias
@@ -0,0 +1 @@
+add de_DE-1901.multi
